bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/config/Device.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mqtt.homeassistant.internal.config;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.google.gson.annotations.JsonAdapter;
+
+/**
+ * Device configuration
+ *
+ * @author Jochen Klein - Initial contribution
+ */
+@NonNullByDefault
+public class Device {
```

Comment:
DTOs shouldn't be annotated with NonNullByDefault, because it gives a false sense of safety as the JSON serializer could set fields to null.

You could move this into a package called `dto` or append `DTO` to the class name to get rid of the checkstyle warning about missing NonNullByDefault annotation.

Reply:
@NonNullByDefault removed, config.dto package added.